### PR TITLE
Kernel alpha

### LINKED ISSUE #2: Amélioration sur le schéma de message
https://github.com/G7r7/message-specs/blob/ab12f957aacd263c98806b25af69bf8bd34190fa/question_example.xml#L41
Je ne pense pas que ce block soit nécessaire. Il serait plus intéressant de pouvoir placer les \<reply> n'importe où dans le message pour laisser l'utilisateur décider de la structure.

https://github.com/G7r7/message-specs/blob/ab12f957aacd263c98806b25af69bf8bd34190fa/question_example.xml#L13-L17
Voir si c'est possible de permettre soit :
 - 1 \<datepicker>
 - plusieurs \<checkbox>
 - plusieurs \<radio_button>
Si c'est possible, on laisse comme ça (c'est léger, c'est bien), sinon on ajoute des \<radio_group> et \<checkbox_group>.

https://github.com/G7r7/message-specs/blob/ab12f957aacd263c98806b25af69bf8bd34190fa/question_example.xml#L19-L22
Le padding dans les id n'est pas nécessaire. On peut simplement utiliser "1", "32", "903".
De plus, si on utilise un système de namespace, on peut se permettre de répéter les id des questions. On peut alors avoir l'id "1" dans le namespace global et l'id "1/1" pour la première question du formulaire "1"

### PR DESCRIPTION
**Pull request by G7r7**

### CODE REVIEW

File: schemas/kernel/question.xsd [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
-                <xsd:element ref="radio"/>
+            <xsd:choice minOccurs="1" maxOccurs="1">
+                <xsd:element ref="checkbox-group"/>
+                <xsd:element ref="colorpicker"/>
+                <xsd:element ref="radio-group"/>
+                <xsd:element ref="datepicker"/>
+                <xsd:element ref="text-input"/>
             </xsd:choice>
+            <xsd:attribute name="id" use="required"/>
+            <xsd:attribute name="text" use="required"/>
```
Comment (DamDC91) [inline schemas/kernel/question.xsd:18]: Missing types for these attributes?

File: schemas/kernel/form.xsd [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+    <xsd:include schemaLocation="form.xsd"/>
     <xsd:element name="form">
         <xsd:complexType>
             <xsd:choice minOccurs="1" maxOccurs="unbounded">
                 <xsd:element ref="question"/>
+                <xsd:element ref="text"/>
+                <xsd:element ref="form"/>
             </xsd:choice>
+            <xsd:attribute name="id" use="required"/>
             <xsd:attribute name="title"/>
```
Comment (DamDC91) [inline schemas/kernel/form.xsd:14]: Missing types for these attributes?

File: schemas/kernel/reply.xsd [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <xsd:element name="reply">
+        <xsd:complexType>
+            <xsd:choice minOccurs="1" maxOccurs="1">
+                <xsd:element ref="reply-checkbox" />
+                <xsd:element ref="reply-radio" />
+                <xsd:element ref="reply-color" />
+                <xsd:element ref="reply-date" />
+                <xsd:element ref="text" />
+            </xsd:choice>
+            <xsd:attribute name="question-id" use="required"/>
```
Comment (DamDC91) [inline schemas/kernel/reply.xsd:17]: Missing type?

File: schemas/kernel/reply-checkbox.xsd [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+    <xsd:element name="reply-checkbox">
+        <xsd:complexType>
+            <xsd:choice minOccurs="0" maxOccurs="unbounded">
+                <xsd:element name="checked-box">
+                    <xsd:complexType>
+                        <xsd:attribute name="offset" use="required"/>
```
Comment (DamDC91) [inline schemas/kernel/reply-checkbox.xsd:8]: missing type

File: schemas/kernel/reply-radio.xsd [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+    <xsd:element name="reply-radio">
+        <xsd:complexType>
+            <xsd:attribute name="offset" use="required"/>
```
Comment (DamDC91) [inline schemas/kernel/reply-radio.xsd:5]: Missing type

File: schemas/kernel/text-input.xsd [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-    <xsd:element name="textinput">
+    <xsd:element name="text-input">
         <xsd:complexType>
             <xsd:attribute name="label"/>
```
Comment (DamDC91) [inline schemas/kernel/text-input.xsd:5]: Missing type

File: schemas/kernel/datepicker.xsd [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+    <xsd:element name="datepicker">
+        <xsd:complexType>
+        </xsd:complexType>
```
Comment (DamDC91) [inline schemas/kernel/datepicker.xsd:5]: useless complexType?

File: schemas/kernel/colorpicker.xsd [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+    <xsd:element name="colorpicker">
+        <xsd:complexType>
+        </xsd:complexType>
```
Comment (DamDC91) [inline schemas/kernel/colorpicker.xsd:5]: Useless?

Comment (G7r7) [reply schemas/kernel/colorpicker.xsd:5]: For now yes!

File: schemas/kernel/datepicker.xsd (hunk 7 above)
Comment (G7r7) [reply schemas/kernel/datepicker.xsd:5]: For now yes!

Comment (DamDC91) [reply schemas/kernel/datepicker.xsd:5]: Will it not be always empty?

It seems that a shorten version exists: `<xsd:complexType/>`

**Comment by fpotier** [comment]
great work @G7r7 and @DamDC91

**Pull request merged**